src/hooks/useStomp.ts

```diff
@@ -0,0 +1,160 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import {
+  Client,
+  StompConfig,
+  StompHeaders,
+  StompSubscription,
+} from '@stomp/stompjs';
+
+/**
+ * USAGE
+ *
+ * connect to the socket URL by const {} = useStomp({ brokerURL: SIMULATION_SOCKET_URL })
+ * subscribe to the topic
+ * publish SUBSCRIBE or UNSUBSCRIBE to the topic
+ */
+
+export const SIMULATION_SOCKET_URL = 'ws://localhost:8080/simulation-socket';
+export const SIMULATION_DATA_TOPIC = '/topic/simulation-data';
+
+// the data returned from the server for each car
+export type VehicleData = {
+  id: string;
+  position: [number, number]; // [x, y]
+  color: string;
+  acceleration: number;
+  speed: number;
+  lane: [number, string]; // [index, id]
+};
+
+let stompClient: Client;
+const subscriptions: { [key: string]: StompSubscription } = {};
```

Comment:
I think this is a lot cleaner, easier to read as well

```suggestion
const subscriptions: Record<string, StompSubscription> = {};
```